freqtrade/optimize/hyperopt_tools.py

```diff
@@ -106,11 +106,20 @@ def print_epoch_details(results, total_epochs: int, print_json: bool,
             HyperoptTools._params_pretty_print(params, 'trailing', "Trailing stop:")
 
     @staticmethod
-    def _params_update_for_json(result_dict, params, space: str) -> None:
+    def _params_update_for_json(result_dict, params, non_optimized, space: str) -> None:
         if space in params:
             space_params = HyperoptTools._space_params(params, space)
+            space_non_optimized = HyperoptTools._space_params(non_optimized, space)
+            all_space_params = space_params
+
+            # Include non optimized params if there are any
+            if len(space_non_optimized) > 0:
+                for non_optimized_param in space_non_optimized:
+                    if non_optimized_param not in all_space_params:
+                        all_space_params[non_optimized_param] = space_non_optimized[non_optimized_param]
```

Comment:
i'd think you should get away with `deep_merge_dicts` here (with non-optimized as first parameter) ... which should simplify this logic (but obviously you need to test if the results are as expected)